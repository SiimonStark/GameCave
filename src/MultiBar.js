import React, { Component } from 'react';

export default class MultiBar extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }
  handleToggle = () => {
    if (this.props.multiplayerToggle === null) { this.props.updateState({ multiplayerToggle: true }) } 
    else if (this.props.multiplayerToggle === true) { this.props.updateState({ multiplayerToggle: false }) }
    else { this.props.updateState({ multiplayerToggle: null }) }
  }
  render() {
    let multiplayerStatus = "ANY";
    if (this.props.multiplayerToggle) { multiplayerStatus = "TRUE" }
    if (this.props.multiplayerToggle === false) { multiplayerStatus = "FALSE" };
    return (
      <div className="multi-bar search-section">
          <button className="button multi-all" onClick={this.handleToggle}> MULTIPLAYER
            <p>{multiplayerStatus}</p>
          </button>
      </div>
    )
  }
}