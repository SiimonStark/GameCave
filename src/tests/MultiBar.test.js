import React from 'react';
import ReactDOM from 'react-dom';
import MultiBar from '../MultiBar';
import { shallow } from 'enzyme';

const updateStateMock = jest.fn();

describe('MultiBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <MultiBar 
      updateState={updateStateMock}/>
    );
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke updateState when button clicked', () => {
    wrapper.find('.button').simulate('click');
    expect(updateStateMock).toBeCalled()
  })
})