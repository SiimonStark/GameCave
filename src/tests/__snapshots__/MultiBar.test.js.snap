// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiBar should match snapshot 1`] = `
<div
  className="multi-bar search-section"
>
  <button
    className="button multi-all"
    onClick={[Function]}
  >
     MULTIPLAYER
    <p>
      ANY
    </p>
  </button>
</div>
`;
